services-camel/src/main/java/edu/unc/lib/cdr/ReplicationProcessor.java

```diff
@@ -0,0 +1,218 @@
+ * It checksums the remote file to make sure it's the same file that was originally ingested.
+ * 
+ * @author lfarrell
+ *
+ */
+public class ReplicationProcessor implements Processor {
+	private static final Logger log = LoggerFactory.getLogger(ReplicationProcessor.class);
+	
+	private final String[] replicationLocations;
+	private final int maxRetries;
+	private final long retryDelay;
+	
+	public ReplicationProcessor(String replicationLocations, int maxRetries, long retryDelay) {
+		this.replicationLocations = splitReplicationLocations(replicationLocations);
+		this.maxRetries = maxRetries;
+		this.retryDelay = retryDelay;
+		
+		checkReplicationLocations(this.replicationLocations);
+	}
+	
+	@Override
+	public void process(Exchange exchange) throws Exception {
+		final Message in = exchange.getIn();
+		
+		String binaryPath = (String) in.getHeader(CdrBinaryPath);
+		String binaryChecksum = (String) in.getHeader(CdrBinaryChecksum);
+		String binaryMimetype = (String) in.getHeader(CdrBinaryMimeType);
+		String binaryUri = (String) in.getHeader(CdrBinaryUri);
+		
+		int retryAttempt = 0;
+		
+		while (true) {
+			try {
+				replicate(binaryPath, binaryChecksum, replicationLocations, binaryMimetype, binaryUri);
+				// Pass mime type and checksum headers along to enhancements
+				exchange.getOut().setHeaders(in.getHeaders()); 
+				break;
+			} catch (Exception e) {
+				if (retryAttempt == maxRetries) {
+					throw e;
+				}
+
+				retryAttempt++;
+				TimeUnit.MILLISECONDS.sleep(retryDelay);
+			}
+		}
+	}
+	
+	private String[] splitReplicationLocations(String replicationLocations) {
+		return replicationLocations.split(";");
+	}
+	
+	private boolean checkReplicationLocations(String[] replicationPaths) {
+		for (String replicationPath : replicationPaths) {
+			if (!Files.exists(Paths.get(replicationPath))) {
+				throw new ReplicationDestinationUnavailableException(String.format("Unable to find replication destination %s", replicationPath));
+			}
+		}
+		
+		return true;
+	}
+	
+	private String createFilePath(String basePath, String originalFileChecksum) {
+		String[] tokens = Iterables.toArray
+				(Splitter.fixedLength(2).split( originalFileChecksum), 
+						String.class);
+		
+		String remotePath = new StringJoiner("/")
+			.add(basePath)
+			.add(tokens[0])
+			.add(tokens[1])
+			.add(tokens[2])
+			.toString();
+		
+		return remotePath;
+	}
+	
+	private String createRemoteSubDirectory(String baseDirectory, String binaryChecksum) {
+		String replicationPath = createFilePath(baseDirectory, binaryChecksum);
+		Path fullPath = Paths.get(replicationPath);
+		
+		if (!Files.exists(fullPath)) {
+			new File(replicationPath).mkdirs();
+		}
+
+		return replicationPath;
+	}
+	
+	private String dbFilePath(String uri, String BinaryChecksum) {
+		String filePath = "/tmp/" + BinaryChecksum;
+		InputStream response = null;
+		
+		try {
+			URLConnection connection = new URL(uri).openConnection();
```

Comment:
You'll probably run into authentication issues with a vanilla java client like this eventually.  Please use FcrepoClient instead to make a GET request, there is probably already one initialized in services-camel's context files that you can pass in.  Using it would be something along the lines of https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/fcrepo4/fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectDataLoader.java#L131